float32 lfw
float32 lmw
float32 lrw
float32 rfw
float32 rmw
float32 rrw
